### PR TITLE
PA-2255 Enhanced Referral Process Agency Responses

### PR DESCRIPTION
**Pull request by Fosol**
User can now select specific agencies and add their responses to projects within ERP.

image

### CODE REVIEW

**Review by jdtoombs — APPROVED** [review]
Looks good!

File: frontend/src/features/projects/erp/forms/AgencyInterest.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  );
+  const agencies = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+    return lookupCode.type === API.AGENCY_CODE_SET_NAME;
+  });
+  const agencyOptions = (agencies ?? []).map(c => mapLookupCodeWithParentString(c, agencies));
+
+  const onAddAgency = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+    event.preventDefault();
+    const selectedAgency = getIn(values, 'addAgencyResponse');
+    const agency = agencies.find(a => a.id == selectedAgency?.value);
```
Comment (jdtoombs) [inline frontend/src/features/projects/erp/forms/AgencyInterest.tsx:39]: `==` -> `===`

Comment (Fosol) [reply frontend/src/features/projects/erp/forms/AgencyInterest.tsx]: Actually this was intentional, but looks like our linting won't allow it.  Essentially `a.id` in typescript is a string, but during runtime it is a number.  Which means `===` will never be `true`...  I'll modify to handle this and make the code way harder to read...

File: frontend/src/features/projects/common/components/FormikTable.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ const FormikTable: React.FC<IFormikTableProps> = ({
   disabled,
 }: IFormikTableProps) => {
   const { values } = useFormikContext();
-  const data = getIn(values, field);
+  const data = getIn(values, field) ?? [];
```
Comment (Fosol) [inline frontend/src/features/projects/common/components/FormikTable.tsx:24]: Fixes unit-tests where `data = null | undefined`

File: frontend/src/features/projects/erp/steps/ErpStep.tsx [hunk 3]
```diff
@@ -50,8 +49,7 @@ const ErpStep = ({ formikRef }: IStepProps) => {
   const currentTab = useSelector<RootState, string | null>(state => state.erpTab) ?? defaultTab;
   const dispatch = useDispatch();
   const history = useHistory();
-  const { projectAgencyResponses } = useAgencyResponseTable();
-  const initialValues: IProject = { ...project, projectAgencyResponses: projectAgencyResponses };
```
Comment (Fosol) [inline frontend/src/features/projects/erp/steps/ErpStep.tsx:54]: Don't populate the agency responses this way

**Review by agalbrai — COMMENTED** [review]
That'll be way better, no more scrolling

**Pull request merged**